File: 프로그래머스/lv1/17681. ［1차］ 비밀지도/［1차］ 비밀지도.java
class Solution {
    static int n;
    public String toBinary(int num) {
        String res = "";
        while (num != 0) {
            res += num % 2;
            num /= 2;
        }
        while(res.length() < n) res += '0';
        return new StringBuffer(res).reverse().toString();
    }

    static char[][] map = new char[16][16];

    public String[] solution(int n, int[] arr1, int[] arr2) {
        Solution.n = n;
        String[] answer = new String[n];
        for (int j = 0; j < n; j++) {
            String s = toBinary(arr1[j]);
            String s2 = toBinary(arr2[j]);
            String temp = "";
            for (int i = 0; i < n; i++) {
                if (s.charAt(i) == '1' || s2.charAt(i) == '1') temp += '#';
                else temp += ' ';
            }
            answer[j] = temp;
        }
        return answer;
    }
}

File: 프로그래머스/1/17681. ［1차］ 비밀지도/［1차］ 비밀지도.cpp
#include <string>
#include <vector>
#include <algorithm>
using namespace std;
int map[16][16],n;

string parse(int k){
    string res="";
    while(k){
        res+=to_string(k%2);
        k/=2;
    }
    int len = res.length();
    for(int i=0;i<n-len;i++){
        res+="0";
    }
    reverse(res.begin(),res.end());
    return res;
}

vector<string> solution(int n, vector<int> arr1, vector<int> arr2) {
    vector<string> answer;
    ::n=n;
    for(int i=0;i<arr1.size();i++){
        string binary=parse(arr1[i]);
        for(int j=0;j<n;j++){
            if(binary[j]=='1'){
                map[i][j]=1;
            }
        }
    }
    for(int i=0;i<arr2.size();i++){
        string binary=parse(arr2[i]);
        for(int j=0;j<n;j++){
            if(binary[j]=='1'){
                map[i][j]=1;
            }
        }
    }
    for(int i=0;i<n;i++){
        string s="";
        for(int j=0;j<n;j++){
            if(map[i][j]){
                s+="#";
            }else{
                s+=" ";
            }
        }
        answer.push_back(s);
    }
    return answer;
}